packages/shared/src/porter.ts

```diff
@@ -6,33 +6,69 @@ import {
   RetrievalKit,
   TreasureMap,
 } from '@nucypher/nucypher-core';
-import axios, { AxiosResponse } from 'axios';
+import axios, {
+  AxiosRequestConfig,
+  AxiosResponse,
+  HttpStatusCode,
+} from 'axios';
 import qs from 'qs';
 
 import { Base64EncodedBytes, ChecksumAddress, HexEncodedBytes } from './types';
 import { fromBase64, fromHexString, toBase64, toHexString } from './utils';
 
-const porterUri: Record<string, string> = {
-  mainnet: 'https://porter.nucypher.community',
-  tapir: 'https://porter-tapir.nucypher.community',
-  oryx: 'https://porter-oryx.nucypher.community',
-  lynx: 'https://porter-lynx.nucypher.community',
+const defaultPorterUri: Record<string, string> = {
+  mainnet: 'https://porter.nucypher.io',
+  tapir: 'https://porter-tapir.nucypher.io',
+  lynx: 'https://porter-lynx.nucypher.io',
 };
 
-export type Domain = keyof typeof porterUri;
+const porterUriSource: string =
+  'https://raw.githubusercontent.com/nucypher/nucypher-porter/development/porter_instances.json';
+
+export type Domain = keyof typeof defaultPorterUri;
+export type PorterURISourceResponse = Record<string, string[]>;
 
 export const domains: Record<string, Domain> = {
   DEVNET: 'lynx',
   TESTNET: 'tapir',
   MAINNET: 'mainnet',
 };
 
-export const getPorterUri = (domain: Domain): string => {
-  const uri = porterUri[domain];
+export const getPorterUri = async (domain: Domain): Promise<string> => {
+  return (await getPorterUris(domain))[0];
+};
+
+export const getPorterUris = async (
```

Comment:
This function is guaranteed to never return an empty array, right? WDYT about adding an `if` statement and throwing an error before we return, just to make sure?

Reply:
I think it will be overkill, there is already error in case there is no default URI so it feels too much